### DISCUSSION TITLE
How to Create Custom Subdirectories with Conveyor

### DISCUSSION THREAD

**Discussion by guiyanakuang** [hydraulic-software/conveyor#111] [Q&A]
I want to create a custom subdirectory custom to map some special files. I found some methods for macOS and Linux as follows:

```
app {

  mac {

    bundle-extras += "composeApp/custom/mac_file" -> "custom/file"
  }

  linux {
    install-path = /usr/lib/projectName

    root-inputs += "composeApp/custom/linux_file" -> "/usr/lib/projectName/custom/file"
  }

  windows {
    # ??
  }
}

```

However, I can't achieve this on Windows. Using `app.inputs += "composeApp/custom/win_file" -> "custom/file"` results in app/custom/file instead of creating a directory parallel to the app directory.

**Comment by mikehearn** [comment]
You're right that there isn't a config key to do this unfortunately. Adding one is straightforward enough, we'll try to get that added for the next release.

May I ask why this is needed? Is it blocking functionality? For macOS and Linux there are cases where files need to be in very specific positions inside the package, but we aren't aware of such cases on Windows which is why we were never asked for this before I guess.

**Reply by guiyanakuang** [reply]
This is not blocking functionality; I can map these files to the same directory as the jar package. However, I believe that allowing some files to be mapped to the same relative path across different platforms would make the code clearer and easier to maintain.
